src/LiveQuery/QueryTools.js

```diff
@@ -99,8 +99,7 @@ function contains(haystack: Array, needle: any): boolean {
       if (typeof ptr === 'string' && ptr === needle.objectId) {
         return true;
       }
-      if (typeof object !== 'undefined' &&
-          ptr.className === needle.className &&
+      if (ptr.className === needle.className &&
```

Comment:
that seems good, you've already validated that needle is not undefined in the conatining if.

Reply:
not really sure if we should be forcing 

`typeof needle !== 'undefined'`

over 

`if(needle)`

I use the second form in my code.  Obviously could be an issue in the case of boolean but i just ignore that...

Reply:
> Obviously could be an issue in the case of boolean but i just ignore that...

if it's a boolean, we'll go to the 'indexOf' case which is correct :)

Reply:
yup, i was musing out loud more generally about if we should enforce a more formal test than just `if(foo)` without advocating that we switch to it.... :).  so safe to ignore me.

Reply:
all those bottom values in JS makes it difficult to make it consistent :/